server/services/challenge.js

```diff
@@ -0,0 +1,58 @@
+import debug from 'debug';
+import { pickBy } from 'lodash';
+import { Observable } from 'rx';
+
+import { cachedMap, getMapForLang, getChallenge } from '../utils/map';
+import { shapeChallenges } from '../../common/app/redux/utils';
+
+const log = debug('fcc:services:challenge');
+const isDev = debug.enabled('fcc:*');
+
+export default function getChallengesForBlock(app) {
+  const challengeMap = cachedMap(app.models);
+  return {
+    name: 'challenge',
+    read: function readChallengesForBlock(
+        req,
+        resource,
+        { dashedName, blockName, lang = 'en' } = {},
+        config,
+        cb
+      ) {
+      const getChallengeBlock$ = challengeMap.map(getMapForLang(lang))
+        .flatMap(({
+          result: { superBlocks },
+          entities: {
+            block: fullBlockMap,
+            challenge: challengeMap
+          }
+        }) => {
+          log(`sourcing challenges for the ${blockName} block`);
```

Comment:
The indenting here for the rest of the function body is off

Reply:
Fixed